js/ui/interaction.js

```diff
@@ -1,325 +0,0 @@
-'use strict';
-
-var handlers = {
-    scrollZoom: require('./handler/scroll_zoom'),
-    boxZoom: require('./handler/box_zoom'),
-    dragRotate: require('./handler/drag_rotate'),
-    dragPan: require('./handler/drag_pan'),
-    keyboard: require('./handler/keyboard'),
-    doubleClickZoom: require('./handler/dblclick_zoom'),
-    touchZoomRotate: require('./handler/touch_zoom_rotate')
-};
-
-var DOM = require('../util/dom'),
-    util = require('../util/util'),
-    Point = require('point-geometry');
-
-module.exports = Interaction;
-
-function Interaction(map) {
-    this._map = map;
-    this._el = map.getCanvasContainer();
-
-    for (var name in handlers) {
-        map[name] = new handlers[name](map);
-    }
-
-    util.bindHandlers(this);
-}
-
-Interaction.prototype = {
-    enable: function () {
-        var options = this._map.options,
-            el = this._el;
-
-        for (var name in handlers) {
-            if (options[name]) this._map[name].enable();
-        }
-
-        el.addEventListener('mousedown', this._onMouseDown, false);
-        el.addEventListener('mouseup', this._onMouseUp, false);
-        el.addEventListener('mousemove', this._onMouseMove, false);
-        el.addEventListener('touchstart', this._onTouchStart, false);
-        el.addEventListener('touchend', this._onTouchEnd, false);
-        el.addEventListener('touchmove', this._onTouchMove, false);
-        el.addEventListener('touchcancel', this._onTouchCancel, false);
-        el.addEventListener('click', this._onClick, false);
-        el.addEventListener('dblclick', this._onDblClick, false);
-        el.addEventListener('contextmenu', this._onContextMenu, false);
-    },
-
-    disable: function () {
-        var options = this._map.options,
-            el = this._el;
-
-        for (var name in handlers) {
-            if (options[name]) this._map[name].disable();
-        }
-
-        el.removeEventListener('mousedown', this._onMouseDown);
-        el.removeEventListener('mouseup', this._onMouseUp);
-        el.removeEventListener('mousemove', this._onMouseMove);
-        el.removeEventListener('touchstart', this._onTouchStart);
-        el.removeEventListener('touchend', this._onTouchEnd);
-        el.removeEventListener('touchmove', this._onTouchMove);
-        el.removeEventListener('touchcancel', this._onTouchCancel);
-        el.removeEventListener('click', this._onClick);
-        el.removeEventListener('dblclick', this._onDblClick);
-        el.removeEventListener('contextmenu', this._onContextMenu);
-    },
```

Comment:
Users can still disable all the handlers by name if necessary. 

```js
map.scrollZoom.disable();
map.boxZoom.disable();
map.dragRotate.disable();
map.dragPan.disable();
map.keyboard.disable();
map.doubleClickZoom.disable();
map.touchZoomRotate.disable();
```

This method was never publicly documented or used internally, so I axed it. Easy to re-add if needed.